support/ebpf/bpfdefs.h

```diff
@@ -4,6 +4,19 @@
 #include "bpf_map.h"
 #include "kernel.h"
 
+// BPF_RODATA_VAR declares a global variable in the .rodata section,
+// ensuring it's not optimized away by the compiler or linker.
+//
+// Arguments:
+//   _type: The data type of the variable (e.g., u32, int, struct my_config).
+//   _name: The name of the global variable.
+//   _value: The initial value for the variable.
+#define BPF_RODATA_VAR(_type, _name, _value)                                                       \
+  _type _name __attribute__((section(".rodata.var"), used)) = _value;
+
```

Comment:
I thought this fixed everything. But sadly this still causes with `coredump` build:
> \# go.opentelemetry.io/ebpf-profiler/tools/coredump
> /tmp/ccFdbFPF.s: Assembler messages:
> /tmp/ccFdbFPF.s:25182: Warning: setting incorrect section attributes for .rodata.var

How about moving this down under the `TESTING_COREDUMP` blocks. And define it as regular variable without section attributes if `TESTING_COREDUMP` is defined?